File: code_provenance/detectors/detector.py
"""Detector registry for Code Provenance."""

from __future__ import annotations

from code_provenance.types import Detector


def create_detector_registry() -> list[Detector]:
    """Create an empty detector registry.

    Registry of all active detectors.
    Populated by the analyzer -- detectors are registered at startup.
    """
    return []

File: src/detectors/detector.ts
import type { Detector } from '../types.js';

// Re-export the Detector interface for convenience
export type { Detector } from '../types.js';

/**
 * Registry of all active detectors.
 * Populated by the analyzer — detectors are registered at startup.
 */
export function createDetectorRegistry(): Detector[] {
  return [];
}
